src/nupic/regions/BacktrackingTMRegion.cpp

```diff
@@ -0,0 +1,1261 @@
+          "",                               // constraints
+          "32",                             // defaultValue
+          ParameterSpec::ReadWriteAccess)); // access
+
+  ns->parameters.add(
+      "maxSegmentsPerCell", // not in Cells4
+      ParameterSpec(
+          "(int) The maximum number of segments allowed on a "
+          "cell. This is used to turn on 'fixed size CLA' mode. When in "
+          "effect, 'globalDecay' is not applicable and must be set to 0 and "
+          "'maxAge' must be set to 0. When this is used (> 0), "
+          "'maxSynapsesPerSegment' must also be > 0. ",
+          NTA_BasicType_Int32,              // type
+          1,                                // elementCount
+          "",                               // constraints
+          "-1",                             // defaultValue
+          ParameterSpec::ReadWriteAccess)); // access
+
+  ns->parameters.add(
+      "maxSynapsesPerSegment", // not in Cells4
+      ParameterSpec(
+          "(int) The maximum number of synapses allowed in "
+          "a segment. This is used to turn on 'fixed size CLA' mode. When in "
+          "effect, 'globalDecay' is not applicable and must be set to 0, and "
+          "'maxAge' must be set to 0. When this is used (> 0), "
+          "'maxSegmentsPerCell' must also be > 0.",
+          NTA_BasicType_Int32,              // type
+          1,                                // elementCount
+          "",                               // constraints
+          "-1",                             // defaultValue
+          ParameterSpec::ReadWriteAccess)); // access
+
+  ns->parameters.add(
+      "outputType", // not in Cells4
+      ParameterSpec(
+          "(string) Can be one of the following (default 'normal'):\n"
+          " - 'normal': output the OR of the active and predicted state. \n"
+          " - 'activeState': output only the active state. \n"
+          " - 'activeState1CellPerCol': output only the active state, and at "
+          "most 1 cell/column. If more than 1 cell is active in a column, the "
+          "one  with the highest confidence is sent up.  ",
+          NTA_BasicType_Byte,               // type
+          0,                                // elementCount
+          "",                               // constraints
+          "normal",                         // defaultValue
+          ParameterSpec::CreateAccess)); // access
+
+  ///////// Parameters not part of the calling arguments //////////
+  ns->parameters.add(
+      "inputWidth",
+      ParameterSpec("(int) width of bottomUpIn. Will be 0 if no links.",
+                    NTA_BasicType_UInt32, // type
+                    1,                    // elementCount
+                    "",                   // constraints
+                    "",                   // defaultValue
+                    ParameterSpec::ReadOnlyAccess)); // access
+
+  //ns->parameters.add(
+  //    "predictedSegmentDecrement",
```

Comment:
Because that parameter is in TM, not BacktrackingTM.  In Python, the BacktrackingTM does 6 flavors of TM. For cpp version of TM it takes a subset of the parameters and calls our TM if some flag is set.  I split them out into two independent plugins and separated the parameters.  I can remove those comments now that I have both written.